components/dashboard/src/onboarding/StepOrgInfo.tsx

```diff
@@ -87,6 +91,27 @@ export const StepOrgInfo: FC<Props> = ({ user, onComplete }) => {
     );
 
     const handleSubmit = useCallback(async () => {
+        // create an org if the user is not a member of one already
+        if (!currentOrg.data) {
```

Comment:
Is this true in Dedicated as well?
Do we need to guard this with `!user.organizationId`?
Why not place this logic closer to user creation, e.g. on initial signup in the backend?

Reply:
Yes, it is true and I put it here because at user creation we don't know if the user is added to an org afterward. disentangle creation of user from creation of org and membership on the API level makes the services a less dependent on each other. I plan to revisit the "join org" so that users who are created and then added to the org will not get an additional synthetic org.

Reply:
> disentangle creation of user from creation of org and membership on the API level makes the services not dependent on each other

While this is true, it also makes the system less stable. But fine with moving in that direction! :+1:

Reply:
You mean if we cannot create an org? Users without an org land on the create org page.